src/test/java/org/broadinstitute/hellbender/engine/dataflow/transforms/RefBasesFromAPITest.java

```diff
@@ -0,0 +1,110 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.coders.CollectionCoder;
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import com.google.cloud.dataflow.sdk.testing.DataflowAssert;
+import com.google.cloud.dataflow.sdk.testing.TestPipeline;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.genomics.v1.Read;
+import htsjdk.samtools.SAMRecord;
+import org.broadinstitute.hellbender.engine.dataflow.DataflowTestData;
+import org.broadinstitute.hellbender.engine.dataflow.coders.GATKReadCoder;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.*;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.dataflow.DataflowUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+public final class RefBasesFromAPITest extends BaseTest {
+
+    @DataProvider(name = "bases")
+    public Object[][] bases(){
+        Object[][] data = new Object[2][];
+        List<Class<?>> classes = Arrays.asList(Read.class, SAMRecord.class);
+        for (int i = 0; i < classes.size(); ++i) {
+            Class<?> c = classes.get(i);
+            DataflowTestData testData = new DataflowTestData(c);
```

Comment:
I did some refactoring as an example of what i think would be cleaner since this pattern is used in a few places

In DataflowTestData
```
    private static final List<Class<?>> GATKReadClasses = Arrays.asList(SAMRecord.class, Read.class);

    public static Object[][] forEachReadClass(Function<DataflowTestData, Object[]> f){
        return GATKReadClasses.stream()
                .map(DataflowTestData::new)
                .map(f).collect(Collectors.toList())
                .toArray(new Object[][]{});
    }
```
in RefBasesFromAPITest

```
    @DataProvider(name = "bases")
    public Object[][] bases(){
        return DataflowTestData.forEachReadClass(RefBasesFromAPITest::getData);
    }

    private static Object[] getData(DataflowTestData testData) {
        List<KV<ReferenceShard, Iterable<GATKRead>>> kvRefShardiReads = testData.getKvRefShardiReads();
        List<SimpleInterval> intervals = testData.getAllIntervals();
        List<KV<ReferenceBases, Iterable<GATKRead>>> kvRefBasesiReads = testData.getKvRefBasesiReads();
        return new Object[]{kvRefShardiReads, intervals, kvRefBasesiReads};
    }
```

`getData` could use a better name though